### PR TITLE
New method that allows not to save the transactions

### PR DESCRIPTION
**Pull request by Superalex0102**
Sorry about the last push, forgot to merge the changes, but this has everything now.

Ref to https://github.com/BG-Software-LLC/SuperiorSkyblock2/pull/757
Made it like you requested a while ago

### CODE REVIEW

File: API/src/main/java/com/bgsoftware/superiorskyblock/api/island/bank/IslandBank.java [hunk 1]
```diff
@@ -38,6 +38,16 @@
...
 
+    /**
+     * Deposit money into the bank, without taking money from any player.
+     *
+     * @param commandSender The player that deposited the money.
+     * @param amount        The amount to deposit.
+     * @param save          Whenever to log the transaction or not.
+     * @return The transaction details.
+     */
+    BankTransaction depositAdminLogMoney(CommandSender commandSender, BigDecimal amount, boolean save);
```
Comment (OmerBenGera) [inline API/src/main/java/com/bgsoftware/superiorskyblock/api/island/bank/IslandBank.java:49]: Should be named "depositAdminMoney"

Comment (OmerBenGera) [inline API/src/main/java/com/bgsoftware/superiorskyblock/api/island/bank/IslandBank.java:49]: the parameter name "save" is not indicative enough.
You can use "saveTransaction" for example, or something like that.

**Review by OmerBenGera — CHANGES_REQUESTED** [review]

**Comment by OmerBenGera** [comment]
By the way, I know you wanted this feature because your database is getting large.
In latest versions of the API, you can make the database-bridge to not save data to the database.
Try checking this out as well :)

**Comment by Superalex0102** [comment]
> By the way, I know you wanted this feature because your database is getting large. In latest versions of the API, you can make the database-bridge to not save data to the database. Try checking this out as well :)

By using https://github.com/BG-Software-LLC/SuperiorSkyblock2/blob/f8c016f2e34c153c1c5de9e6543d1f165cc2832d/API/src/main/java/com/bgsoftware/superiorskyblock/api/island/bank/IslandBank.java#L39?

**Comment by OmerBenGera** [comment]
> > By the way, I know you wanted this feature because your database is getting large. In latest versions of the API, you can make the database-bridge to not save data to the database. Try checking this out as well :)
> 
> By using
> 
> https://github.com/BG-Software-LLC/SuperiorSkyblock2/blob/f8c016f2e34c153c1c5de9e6543d1f165cc2832d/API/src/main/java/com/bgsoftware/superiorskyblock/api/island/bank/IslandBank.java#L39
> ?

You can change the DatabaseBridge mode (DatabaseBridge is the object that handles interactions with the databases) of the island to "IDLE", which will halt saving of data to database:
https://github.com/BG-Software-LLC/SuperiorSkyblock2/blob/dev/API/src/main/java/com/bgsoftware/superiorskyblock/api/data/DatabaseBridge.java#L74

**Comment by Superalex0102** [comment]
> 

image
and how can I use it? I tried it like above.

**Comment by OmerBenGera** [comment]
> > 
> 
> image and how can I use it? I tried it like above.

The API with the updated methods is not yet published.
You can find the API jar on <https://hub.bg-software.com/> and use it locally until a new version of SSB is released.

**Comment by Superalex0102** [comment]
Thank you, you can close this, I will make changes on WildChests in the future, to include this

**Pull request closed without merging**